userd/userd.go

```diff
@@ -71,17 +50,29 @@ func (ud *Userd) Init() error {
 		return err
 	}
 
-	reply, err := ud.conn.RequestName(busName, dbus.NameFlagDoNotQueue)
-	if err != nil {
-		return err
+	ud.dbusIfaces = []dbusInterface{
+		&Launcher{},
 	}
+	for _, iface := range ud.dbusIfaces {
+		reply, err := ud.conn.RequestName(iface.Name(), dbus.NameFlagDoNotQueue)
+		if err != nil {
+			return err
+		}
 
-	if reply != dbus.RequestNameReplyPrimaryOwner {
-		err = fmt.Errorf("cannot obtain bus name '%s'", busName)
-		return err
-	}
+		if reply != dbus.RequestNameReplyPrimaryOwner {
+			err = fmt.Errorf("cannot obtain bus name '%s'", iface.Name())
+			return err
+		}
 
-	ud.createAndExportInterfaces()
+		var buffer bytes.Buffer
+		buffer.WriteString("<node>")
+		buffer.WriteString(iface.IntrospectionData())
+		buffer.WriteString(introspect.IntrospectDataString)
+		buffer.WriteString("</node>")
```

Comment:
For such simple cases, I suspect this is both simpler and faster than using buffer.String():

`"<node>" + foo + bar "</node>"`

Similar to the other point, this is not a requirement.